src/k-provider/ovp/provider-parser.js

```diff
@@ -344,4 +338,24 @@ export default class OVPProviderParser extends BaseProviderParser {
     }
     return 'https';
   }
+
+  /**
+   * Applies the request host regex on the url
+   * @function _applyRegexAction
+   * @param {Object} playbackContext - The playback context
+   * @param {string} playUrl - The original url
+   * @returns {string} - The request host regex applied url
+   * @static
+   * @private
+   */
+  static _applyRegexAction(playbackContext: Object, playUrl: string): string {
```

Comment:
pass action object directly and move the code to the `KalturaPlaybackContext` so you can get this as a service from it - the context of if it has regex action seems better suited there.